### PR TITLE
kinetic-devel release

### PR DESCRIPTION
**Pull request by AustinDeric**
addresses issue #115

### CODE REVIEW

**Review by gavanderhoorn — CHANGES_REQUESTED** [review]
Nice work, three things:

 1. `industrial_core` has a Kinetic release. Is building from sources really necessary? The `abb.rosinstall` file would not be needed anymore if the released version is sufficent either.
 1. I'm not sure all of these packages need C++11 enabled, and for metapackages (`abb`) we're not allowed to change the `CMakeLists.txt`, so that will need to be reverted.
 1. the build instructions ask the user to create a `abb_experimentnal_ws` workspace, should that be `abb_ws` (also note the typo)? And if we don't use `abb.rosinstall` then the instructions should include a `rosdep install ..` step.

**Comment by AustinDeric** [comment]
1.  Its not necessary. I will also describe the way to build this package using the Debian as well and place this before the source install method.  You will notice that the .travis.yml uses the Debian method to test.  With that said, i would prefer to leave the abb.rosinstall file for those of us who want to keep the dependencies in the local src folder.
2. I thought it may be good to use c++11 going forward, is it ok to leave it?  I will revert the metapackage.
3. typo :(. I will specify the rosdep as the "recommended" method.

**Comment by gavanderhoorn** [comment]
> With that said, i would prefer to leave the abb.rosinstall file for those of us who want to keep the dependencies in the local src folder.

The thing is that ROS best practices are to avoid building from sources as much as possible. We can add instructions on how to build from sources, but it should be made *really* explicit that it's only *ever* needed if you want to use the packages on platforms / OS for which there is no release, you need a feature that hasn't been released yet or you want to work on the packages themselves.

> I thought it may be good to use c++11 going forward, is it ok to leave it?

I completely agree with encouraging people to use C++11, but the only packages in this repository with any C++ sources in them are the IKFast plugins and the driver. And IIRC only the IK plugins really need it enabled (because of MoveIt). So the flag doesn't do anything for the other packages, and as packages in the official repositories are taken as examples of how to set things up, it would be really nice if we can avoid adding things that don't belong. That could lead to cargo cult copy-cat behaviour which I don't think we should want.

> I will specify the rosdep as the "recommended" method.

Great. Much appreciated.

File: README.md [hunk 1]
```diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
...
+## Build
+### Build from Binaries (Recommended)
+This build methods uses:
+ - [catkin tools](https://catkin-tools.readthedocs.io/en/latest/) for building.
+```
+mkdir -p abb_ws/src
+cd abb_ws/src
+git clone -b kinetic-devel https://github.com/ros-industrial/abb.git
+rosdep update
+rosdep install abb
```
Comment (gavanderhoorn) [inline README.md]: May I suggest changing this to `rosdep install --from-paths . --ignore-src`?

File: README.md [hunk 2]
```diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
...
+### Build from Source (Developers Only)
+This build methods uses:
+ - [wstool](http://wiki.ros.org/wstool) for rosinstall.
+ - [catkin tools](https://catkin-tools.readthedocs.io/en/latest/) for building.
+```
+mkdir -p abb_ws/src
+cd abb_ws/src
+git clone -b kinetic-devel https://github.com/ros-industrial/abb.git
+wstool init .
+wstool merge abb/abb.rosinstall
```
Comment (gavanderhoorn) [inline README.md]: The above three lines could be simplified to read:
```
wstool init .
wstool merge https://raw.githubusercontent.com/ros-industrial/abb/kinetic-devel/abb.rosinstall
```

File: README.md [hunk 3]
```diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
...
+wstool update
+catkin build
+```
```
Comment (gavanderhoorn) [inline README.md]: And it's certainly implied, but if we're going to be this detailed about documenting build steps then perhaps we should add a `source devel/setup.bash` as well?

Comment (gavanderhoorn) [reply README.md]: And a `source /opt/ros/kinetic/setup.bash`?

File: README.md (hunk 1 above)
Comment (AustinDeric) [reply README.md]: fixed.

File: README.md (hunk 2 above)
Comment (AustinDeric) [reply README.md]: fixed.

File: README.md (hunk 3 above)
Comment (AustinDeric) [reply README.md]: I'd prefer to end the build instructions at a successful build. An examples section should start where the build sections left off and include sourcing the packages.

**Comment by AustinDeric** [comment]
In addition to the replied fixes, i removed c++11 flags where they were not needed.  I also added more emphasis in the readme that "build from sources"  is not recommended.

**Comment by AustinDeric** [comment]
I took out all references and code related to building from source. @Levi-Armstrong and @gavanderhoorn are we good with the state of this package?  If so, can we create a new branch with it?

Thanks!

**Comment by Levi-Armstrong** [comment]
@AustinDeric I created a kinetic-devel branch can you verify that this PR with the new branch on kinetic?

**Comment by AustinDeric** [comment]
@Levi-Armstrong i rebased this PR using the [ros-industrial/abb/tree/kinetic-devel](https://github.com/ros-industrial/abb/tree/kinetic-devel) branch.

Everything looks good.  Based on the comments in this PR, i will update the [abb_experimental ](https://github.com/ros-industrial/abb_experimental) kinetic-devel PR so we dont have to repeat ourselves.  Can you get the kinetic-devel branch out?

Thank you all for your hard work!

**Pull request merged**